src/bin/pg_dump/pg_dump.c

```diff
@@ -2439,7 +2439,12 @@ dumpTableData(Archive *fout, const TableDataInfo *tdinfo)
 	/* We had better have loaded per-column details about this table */
 	Assert(tbinfo->interesting);
 
-	if (dopt->dump_inserts == 0)
+	if(hasSqlvariantColumn(tbinfo)){
```

Comment:
Let's also add `isBabelfishDatabase` check here or in `hasSqlvariantColumn` function.

Reply:
Added another check in the of condition.